lazy_theta_star_planner/include/lazy_theta_star_planner/lazy_theta_star2.h

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020 Anshumaan Singh
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LAZY_THETA_STAR_B__LAZY_THETA_STAR_H_
+#define LAZY_THETA_STAR_B__LAZY_THETA_STAR_H_
+
+#include <cmath>
+#include <vector>
+#include <queue>
+#include <algorithm>
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+#include "nav2_costmap_2d/cost_values.hpp"
+
+typedef int id;
+typedef double cost;
+typedef int map_pts;
+typedef double world_pts;
+
+#define INF_COST DBL_MAX
+#define LETHAL_COST 252
+
+template<typename coordsT>
```

Comment:
Do you think this is really necessary to template this?

Reply:
You only use double through out the file.

Reply:
I actually use them to store src and dst points too, using int

Reply:
Yeah, but you don't use this templated class for it. 

Overall, I think the templates are really not required anywhere in this code. I haven't seen a location where you make use of it but it certainly makes things look more complex